1284 Journalføring av brukers meldekort

Henreich opened the pull request:
## Beskrivelse
1284 Journalføring av brukers meldekort
For at det bruker sender inn blir journalført og for at bruker skal kunne se hva de har sendt inn i ettertid.

Endringer:
* Lagt til klient for pdfgen - Konverterer brukers innsending til en PDF
* Lagt til klient for dokarkiv - journalfører meldekortet (PDF og JSON) i dokarkiv og blir synlig i Gosys
* Brevkoden som benyttes i arena videreføres da vi ikke har noen ytterligere behov. Det ble foreslått å legge til et postfiks for å beholde nåværende brevkode, men vi så det som unødvendig da man kan skille mellom dokumentene basert på tema for statistikkformål.
* Fordi vi ikke kommer til å prioritere å legge til hjelpefunksjoner i pdfgen (eller pdfgen-core som vi bruker imaget til) så parser vi heller datoene i backend før de sendes til pdfgen for å få det riktig. Langsiktig er det nok bedre å sende ISO-formaterte datoer og ha støttefunksjoner på plass, men det er nedprioritert per nå.

hestad commented on src/main/kotlin/no/nav/tiltakspenger/meldekort/api/Configuration.kt:
```diff
@@ -8,11 +8,18 @@ import com.natpryce.konfig.intType
 import com.natpryce.konfig.stringType
 import no.nav.tiltakspenger.meldekort.api.auth.AzureTokenProvider
+import no.nav.tiltakspenger.meldekort.api.tilgang.Rolle
+import java.util.*
```
Trodde ktlint nekte star imports

thorellefsennav replied:
java.util.* er en special case: https://github.com/pinterest/ktlint/blob/48cc3f368248c837f81476b17384cb1ca75ed394/documentation/snapshot/docs/faq.md?plain=1#L186

hestad reviewed (approved):
lgtm

Henreich committed:
1284 Journalføring av brukers meldekort

For at det bruker sender inn blir journalført og for at bruker skal kunne se hva de har sendt inn i ettertid.

Endringer:
* Lagt til klient for pdfgen - Konverterer brukers innsending til en PDF
* Lagt til klient for dokarkiv - journalfører meldekortet (PDF og JSON) i dokarkiv og blir synlig i Gosys
* Brevkoden som benyttes i arena videreføres da vi ikke har noen ytterligere behov. Det ble foreslått å legge til et postfiks for å beholde nåværende brevkode, men vi så det som unødvendig da man kan skille mellom dokumentene basert på tema for statistikkformål.
* Fordi vi ikke kommer til å prioritere å legge til hjelpefunksjoner i pdfgen (eller pdfgen-core som vi bruker imaget til) så parser vi heller datoene i backend før de sendes til pdfgen for å få det riktig. Langsiktig er det nok bedre å sende ISO-formaterte datoer og ha støttefunksjoner på plass, men det er nedprioritert per nå.

tiafiring commented on .nais/nais.yml:
```diff
@@ -59,4 +59,8 @@ spec:
   observability:
```
Jeg tror du kanskje må legge til de nye utgående integrasjonene i outbound-policy, tilsvarende som her https://github.com/navikt/tiltakspenger-soknad-api/blob/main/.nais/nais.yml#L52 :)

tiafiring replied:
veldig fint at du la til loki-loggingen forresten! 🏆

tiafiring commented on src/main/kotlin/no/nav/tiltakspenger/meldekort/clients/dokarkiv/JournalpostRequest.kt:
```diff
@@ -0,0 +1,125 @@
+        dokumenter = listOf(
+            JournalpostDokument(
+                tittel = tittel,
+                brevkode = "NAV 00-10.02", // TODO Hva blir riktig brevkode? Hentet fra arena sin journalføring av meldekort, men den tar for seg alle meldekort?
```
Hvis vi skal få et nytt meldekort, kanskje vi må ha en ny brevkode..? 🤔 Det kan også hende det er både greit og riktig å bruke den vi har fra før også. Litt usikker på hvem som vil kunne hjelpe oss med den avklaringen, men kanskje #team_dokumentløsninger kan være et bra sted å starte.

Henreich replied:
takk! skal prøve meg der :D

Henreich replied:
startet en tråd https://nav-it.slack.com/archives/C6W9E5GPJ/p1739169711215439

tiafiring commented on src/main/kotlin/no/nav/tiltakspenger/meldekort/clients/KonfigurertHttpClient.kt:
```diff
@@ -0,0 +1,75 @@
+                }
+            level = LogLevel.INFO
+        }
+        expectSuccess = true
```
Hvis jeg skulle ha satt opp httpclient som ikke ble brukt av noe annet ennå hadde jeg nok valgt å sette denne til false. Når den er true vil klienten kaste exceptions (forskjellige avhengig av om det er en 4xx-feil eller 5xx-feil), og feilhåndtering blir litt mer kronglete, hvertfall hvis man er vant til at det ikke kastes exceptions og at man bare kan sjekke på responskoder. 

Jeg vil også anbefale å bruke apache og ikke cio siden det har vært mye tull og minnelekkasjer med cio-klienten :)

Henreich replied:
gode punkter. Jeg lot denne stå litt pga diskusjonen jeg startet opp i #tp-utvikling på slack :D Nå har denne PR-en introdusert både java.net og ktor sin httpclient, så jeg tenker til syvende og sist at jeg bare introduserer en av de.

tiafiring replied:
Regnet nesten litt med det :D Det hadde vært supert om vi kunne klare oss med bare en av dem 🙏

tiafiring reviewed (approved):
🤩
